Extension/src/LanguageServer/referencesModel.ts

```diff
@@ -6,126 +6,277 @@
 import * as vscode from 'vscode';
 import { ReferenceType, ReferenceInfo, ReferencesResult } from './references';
 
-export class Model {
-    readonly FileItems: FileItem[] = [];
-    readonly ReferenceItems: ReferenceItem[] = [];
-    readonly ReferenceTypeItems: ReferenceTypeItem[] = [];
+export type RenameResultCallback = (result: ReferencesResult) => void;
+
+let currentReferencesModel: ReferencesModel;
+
+export class ReferencesModel {
+    readonly nodes: TreeNode[] = []; // Raw flat list of references
+    private renameResultsCallback: RenameResultCallback;
     private originalSymbol: string = "";
+    public groupByFile: boolean;

```

Comment:
>groupByFile [](start = 11, length = 11)

can the view (referencesView) own groupByFile property?

Reply:
Some providers do not group by file (i.e. pending rename tree).  Also, isn't it preferable for the provider to be driven by data in the model?